pkg/channel/fanout/fanout_message_handler.go

```diff
@@ -105,6 +126,7 @@ func SubscriberSpecToFanoutConfig(sub eventingduckv1.SubscriberSpec) (*Subscript
 
 	var deadLetter *url.URL
 	if sub.Delivery != nil && sub.Delivery.DeadLetterSink != nil && sub.Delivery.DeadLetterSink.URI != nil {
+		// TODO: Bug, this does not seem to support refing the Ref field.
```

Comment:
?

Reply:
Ah, i see your point, I recall i looked into this a while ago, and my understanding is that when this function is invoked, we're 100% sure that the ref is resolved, because it's done by the subscription duck controller (or the channel duck controller, i don't remember) before this one is invoked. So it can't happen that this sub.Delivery doesn't contain the url. But I may be wrong, feel free to check it again

Reply:
https://github.com/knative/eventing/issues/4376